src/main/java/org/spongepowered/common/util/SkinUtil.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * This file is part of Sponge, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.common.util;
+
+import org.spongepowered.api.Sponge;
+import org.spongepowered.api.data.type.SkinPart;
+import org.spongepowered.api.util.GuavaCollectors;
+import org.spongepowered.common.data.type.SpongeSkinPart;
+
+import java.util.Set;
+
+public final class SkinUtil {
+
+    private SkinUtil() {
+    }
+
+    public static Set<SkinPart> fromFlags(int flags) {
+        return Sponge.getRegistry().getAllOf(SkinPart.class).stream()
+                .filter(part -> part instanceof SpongeSkinPart)
```

Comment:
Also I thing this check is not neccessary since, we use our own implementation here, without ability to register more.
just use a plain `.map(input -> (SpongeSkinPart) input)`. If there will ever be more, then we would have to add them to our registry anyway.